v0/destinations/marketo/transform.js

```diff
@@ -0,0 +1,290 @@
+const get = require("get-value");
+const axios = require("axios");
+const { EventType } = require("../../../constants");
+const { identifyConfig, formatConfig } = require("./config");
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  getFieldValueFromMessage
+} = require("../../util");
+
+// //////////////////////////////////////////////////////////////////////
+// BASE URL REF: https://developers.marketo.com/rest-api/base-url/
+// //////////////////////////////////////////////////////////////////////
+
+// calls Marketo Auth API and fetches bearer token
+// fails the transformer if auth fails
+// ------------------------
+// Ref: https://developers.marketo.com/rest-api/authentication/#creating_an_access_token
+const getAuthToken = async destination => {
```

Comment:
let's wrap all axios request in try-catch separately and throw that error back to versionrouter?

Reply:
for cases like where resp is empty because of marketo error/ network error etc.